libtelemetry/src/full/java/com/mapbox/android/telemetry/AppStateUtils.java

```diff
@@ -196,11 +207,20 @@ private static AppState arbitrage(AppState stateFromActivityManager,
     return stateFromActivityManager;
   }
 
-  public static AppState getAppState(@NonNull Context context) {
-    LogUtils.v(TAG, "Getting app state...");
-    final AppState state = arbitrage(getAppStateFromActivityManager(context),
-        getLastKnownActivityState(context));
-    LogUtils.v(TAG, "getAppState() returns " + state);
-    return state;
+  public interface GetAppStateCallback {
+    void onReady(AppState state);
+  }
+
+  public static void getAppState(@NonNull final Context context, final GetAppStateCallback callback) {
+    ioExecutor.execute(new Runnable() {
+      @Override
+      public void run() {
+        Log.i(TAG, "Getting app state...");
```

Comment:
Do we want to keep those logs?

Reply:
Ops, by accident, moving back to `Logger.v`.